nemo/core/config/optimizers.py

```diff
@@ -238,6 +240,11 @@ def get_optimizer_config(name: str, **kwargs: Optional[Dict[str, Any]]) -> Optim
         )
 
     scheduler_params = AVAILABLE_OPTIMIZER_PARAMS[name]
+
+    if kwargs is not None and len(kwargs) != 0:
+        kwargs = OmegaConf.create(kwargs)
+        OmegaConf.merge(scheduler_params(), kwargs)
+
     scheduler_params = partial(scheduler_params, **kwargs)
```

Comment:
You can avoid the partial if you want
```python
    scheduler_params = OmegaConf.structured(AVAILABLE_OPTIMIZER_PARAMS[name])
    if kwargs is not None and len(kwargs) != 0:
        kwargs = OmegaConf.create(kwargs)
        scheduler_params = OmegaConf.merge(scheduler_params, kwargs)
    return scheduler_params  # A DictConfig not a (class pointer/definition, not sure what to call this)
```

Reply:
Scheduler parameter dataclasses are somewhat special, they dont hold the `max_steps` kwarg even though their constructor does have this value (in most but not all constructors). I have to defer its calculation until we resolve either `iters_per_batch` or the actual passed `max_steps` and finally instantiate the constructor of the scheduler with the union of the params and this `max_steps`.